stacker/util.py

```diff
@@ -459,19 +477,19 @@ class SourceProcessor():
     """Makes remote python package sources available in the running python
        environment."""
 
-    def __init__(self, stacker_cache_dir=None):
+    def __init__(self, config={}):
```

Comment:
So two things here:

1. You shouldn't use a mutable type (like a dict) as a default argument to a variable, it can lead to some frustrating bugs.
2. I honestly don't really like passing in a generic config dictionary and then relying on the contents - again can lead to some frustrating typo type bugs.

The second one is the most pressing, so I see a couple of options:

1. We pass in specific arguments from the config, rather than using config itself.
2. We pass in an object that has all the correct fields on it.